front/components/vaults/VaultCreateWebsiteModal.tsx

```diff
@@ -1,32 +1,602 @@
+  const [dataSourceNameError, setDataSourceNameError] = useState<string | null>(
+    null
+  );
+
+  const [maxPages, setMaxPages] = useState<number | null>(
+    webCrawlerConfiguration?.maxPageToCrawl || 50
+  );
+  const [maxDepth, setMaxDepth] = useState<DepthOption>(
+    webCrawlerConfiguration ? webCrawlerConfiguration.depth : 2
+  );
+  const [crawlMode, setCrawlMode] = useState<"child" | "website">(
+    webCrawlerConfiguration?.crawlMode || "website"
+  );
+  const [selectedCrawlFrequency, setSelectedCrawlFrequency] =
+    useState<CrawlingFrequency>(
+      webCrawlerConfiguration?.crawlFrequency || "monthly"
+    );
+  const [advancedSettingsOpened, setAdvancedSettingsOpened] = useState(false);
+
+  const existingHeaders = webCrawlerConfiguration?.headers
+    ? Object.entries(webCrawlerConfiguration.headers).map(([key, value]) => {
+        return { key, value };
+      })
+    : [];
+  const [headers, setHeaders] = useState(existingHeaders);
+
+  const { mutateVaultDataSourceViews } = useVaultDataSourceViews({
+    workspaceId: owner.sId,
+    vaultId: vault.sId,
+    category: WEBSITE_CAT,
+  });
+
+  const frequencyDisplayText: Record<CrawlingFrequency, string> = {
+    never: "Never",
+    daily: "Every day",
+    weekly: "Every week",
+    monthly: "Every month",
+  };
+
+  const depthDisplayText: Record<DepthOption, string> = {
+    0: "0 level",
+    1: "1 level",
+    2: "2 levels",
+    3: "3 levels",
+    4: "4 levels",
+    5: "5 levels",
+  };
+
+  useEffect(() => {
+    if (isUrlValid(dataSourceUrl)) {
+      setDataSourceName(urlToDataSourceName(dataSourceUrl));
+    }
+  }, [dataSourceUrl]);
+
+  const validateForm = useCallback(() => {
+    let urlError = null;
+    let nameError = null;
+
+    // Validate URL
+    if (!isUrlValid(dataSourceUrl)) {
+      urlError =
+        "Please provide a valid URL (e.g. https://example.com or https://example.com/a/b/c)).";
+    }
+
+    // Validate Name
+    const nameExists = dataSources.some(
+      (d) => d.name === dataSourceName && d.id !== dataSourceOrView?.id
+    );
+    const dataSourceNameRes = isDataSourceNameValid(dataSourceName);
+    if (nameExists) {
+      nameError = "A Folder with the same name already exists";
+    } else if (!dataSourceName.length) {
+      nameError = "Please provide a name.";
+    } else if (dataSourceNameRes.isErr()) {
+      nameError = dataSourceNameRes.error;
+    }
+
+    setDataSourceUrlError(urlError);
+    setDataSourceNameError(nameError);
+    return !urlError && !nameError;
+  }, [dataSourceName, dataSources, dataSourceUrl, dataSourceOrView?.id]);
+
+  useEffect(() => {
+    if (isSubmitted) {
+      validateForm();
+    }
+  }, [dataSourceName, dataSourceUrl, isSubmitted, validateForm]);
+
+  const handleCreate = async () => {
+    setIsSubmitted(true);
+
+    if (!validateForm()) {
+      return;
+    }
+
+    setIsSaving(true);
+    if (webCrawlerConfiguration === null) {
```

Comment:
Can we split this in two distinct functions, one for website creation and another one for regular update. Also the component name seems off as it seems to allow both creation and update of a website.